### DISCUSSION TITLE
Gatsby slice feature

### DISCUSSION THREAD

**Discussion by Jaykishan0705** [gatsbyjs/gatsby#37952] [Help]
Hello Everyone.

I have few doubts regarding gatsby Slice API.

1. How Gatsby know that specific slice needs to be updated. Asking these because if we are changing any code for that component, then will all the page will be updated or just that slice will update ?
2. Should we use Slice for all the templates? As described in the docs, Gatsby create placeholder for each Slice component, then what is the drawback of using Slice even if that Placeholder is used in only one page.
eg. use-case. I have one website. and on this website I am adding one blog each week. To add one blog, I need to create one page in contentful ( no code change is required. reusing the template for BlogPage ). Should I need to wrap the BlogPage Template with Slice Placeholder. I see one benefit of wrapping the BlogPage template with Slice Placeholder. Just one slice placeholder will create and all the other page will be untouched in gatsby build. please correct me if I am wrong.

**Comment by LekoArts** [comment]
1. If you only update your slice component, then only this HTML file will be rewritten. Then it'll be stitched into the pages that use the slice. If you wouldn't use it as a slice, all pages would be updated.
2. They have restrictions: https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-slice/#restrictions-on-using-slice-placeholder - if that's not relevant to you, feel free to use slices for common parts of your website, e.g. header, footer, etc. (https://www.gatsbyjs.com/docs/how-to/performance/using-slices)
